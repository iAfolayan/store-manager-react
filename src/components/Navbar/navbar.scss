.sidebar {
	background: #0f488a;
	height: 100vh;
	position: fixed;
}

#logo {
	width: 100%;
	height: 20%;
	margin:0 auto;
	background: url('./images/logo.PNG') no-repeat center;
}

nav {
	>ul {
		line-height: 60px;
		padding-inline-start: 0px;
		>li {
			list-style: none;
			padding-inline-start: 2px;
			>a {
				color: #FFF;
			}
		}
	}
}

li#usrManagement, li#prdManagement {
	color: #fff;
	font-size: 1rem;
}
.usrChild {
	>ul {
		padding-inline-start: 0px;
		>li {
			line-height: 30px;
			margin: 0;
			display: block;
			font-size: 12px;
			>a {
				padding-inline-start: 5px !important;
				color: #FFF;
				&:hover {
					color: #FFF;
				}
			}
		}
	}
}
.prdChild {
	>ul {
		padding-inline-start: 0px;
		>li {
			line-height: 30px;
			margin: 0;
			display: block;
			font-size: 12px;
			>a {
				padding-inline-start: 5px !important;
				color: #FFF;
				&:hover {
					color: #FFF;
				}
			}
		}
	}
}
.rptChild {
	>ul {
		padding-inline-start: 0px;
		>li {
			line-height: 30px;
			margin: 0;
			display: block;
			font-size: 12px;
			>a {
				padding-inline-start: 5px !important;
				color: #FFF;
				&:hover {
					color: #FFF;
				}
			}
		}
	}
}

.welcomeMessage {
	color:#fff;
	font-size: 1rem
}