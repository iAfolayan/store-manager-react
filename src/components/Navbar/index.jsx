import React, { Component } from 'react';
import { Menu } from 'semantic-ui-react';
import './navbar.scss';

export default class MenuExampleSizeMini extends Component {
  state = { activeItem: 'home' };

  handleItemClick = (e, { name }) => this.setState({ activeItem: name });

  render() {
    const { activeItem } = this.state;

    return (
      <Menu className="navbar" stackable size="large">
        <Menu.Item />

        <Menu.Item
          name="features"
          active={activeItem === 'features'}
          onClick={this.handleItemClick}
        >
          Features
        </Menu.Item>

        <Menu.Item
          name="testimonials"
          active={activeItem === 'testimonials'}
          onClick={this.handleItemClick}
        >
          Testimonials
        </Menu.Item>

        <Menu.Item name="sign-in" active={activeItem === 'sign-in'} onClick={this.handleItemClick}>
          Sign-in
        </Menu.Item>
      </Menu>
    );
  }
}
