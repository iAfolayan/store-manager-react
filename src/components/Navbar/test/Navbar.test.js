import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '..';

describe('<NavBar />', () => {
  test('should rendered NavBar without crashing', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should simulate logout event', () => {
    const logout = jest.fn();
    const wrapper = shallow(<Navbar />);
    wrapper.instance().logout();
    expect(wrapper).toBeDefined();
  });
});
