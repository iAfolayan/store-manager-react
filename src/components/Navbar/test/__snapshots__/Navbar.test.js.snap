// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should rendered NavBar without crashed 1`] = `
<div
  className="sidebar pt-3"
>
  <img
    src="test-file-stub"
  />
  <h5
    className="d-flex justify-content-center welcomeMessage"
  >
    Welcome. iAfolayan
  </h5>
  <hr />
  <br />
  <nav
    className="navbar"
  >
    <ul>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/products"
        >
          Home
        </NavLink>
      </li>
      <li
        id="usrManagement"
      >
        User Management
        <div
          className="usrChild"
        >
          <ul>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="p-2"
                to="user"
              >
                Create new User
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="p-2"
                to="sales"
              >
                View Sales Record
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="p-2"
                to="users"
              >
                View users
              </NavLink>
            </li>
          </ul>
        </div>
      </li>
      <li
        id="prdManagement"
      >
        Product Management
        <div
          className="prdChild"
        >
          <ul>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="p-2"
                to="category"
              >
                Create Category
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="p-2"
                to="product"
              >
                Create Product
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="p-2"
                to="modify"
              >
                Modify product
              </NavLink>
            </li>
          </ul>
        </div>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          onClick={[Function]}
          to=""
        >
          Logout
        </NavLink>
      </li>
    </ul>
  </nav>
</div>
`;
